### PR TITLE
feat(stdlib/contrib): add teams package

### PR DESCRIPTION
**Pull request by sranka**
This PR adds **teams** package to flux so that notifications can be sent from flux to Microsoft Teams. It is the first step in influxdata/influxdb#17939

Notes:
- the implementation and tests follow the concepts in existing "slack" package
- teams event handler is available in [V1 kapactior](https://github.com/influxdata/kapacitor/pull/1965)
- flux's teams package will be then used in influxdata/influxdb to introduce a new teams notification endpoint type

### Done checklist
- [x] Test cases written
- [x] Sign [CLA](https://www.influxdata.com/legal/cla/) (if not already signed)

### CODE REVIEW

**Comment by rockstar** [comment]
Could you please move this module to use our new "contrib" layout? You can see how to do that [here](https://github.com/influxdata/flux/tree/master/stdlib/contrib).

**Comment by sranka** [comment]
@rockstar it is now refactored to a new contrib layout + I also added a README.md to the new package.

File: libflux/src/core/semantic/types.rs [hunk 1]
```diff
@@ -1144,7 +1144,7 @@ impl Function {
                     // This means they should match. Enforce this condition by inserting
                     // the pipe argument into the required ones with the same key.
                     f.req.insert(fp.k.clone(), fp.v);
-                    g.req.insert(fp.k.clone(), gp.v);
+                    g.req.insert(fp.k, gp.v);
```
Comment (sranka) [inline libflux/src/core/semantic/types.rs:1147]: discovered by clippy, required for the tests to pass

**Comment by sranka** [comment]
@rockstar conflicts resolved, a new rust's clippy recommendation applied, pls merge

**Pull request merged**